addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -3024,6 +3026,23 @@ options.registry.ConditionalVisibility = options.Class.extend({
 
         this._updateCSSSelectors();
     },
+    /**
+     * Selects a value for target's data-attributes.
+     * Should be used instead of selectRecord if the visibility is not related
+     * to database values.
+     *
+     * @see this.selectClass for parameters
+     */
+    selectValue(previewMode, widgetValue, params) {
+        const widgetValueIndex = params.possibleValues.indexOf(widgetValue);
+        if (widgetValue) {
+            this.$target[0].dataset[params.saveAttribute] = JSON.stringify([{value: widgetValue.trim(), id: widgetValueIndex}]);
```

Comment:
Strange you need to trim the value here?

Reply:
No it's useless, I forgot to remove after removing all that 'or' mechanism. Just removed it :)